include/aspect/boundary_temperature/interface.h

```diff
@@ -255,9 +258,26 @@ namespace aspect
          * that matches the given type, return a NULL pointer.
          */
         template <typename BoundaryTemperatureType>
+        DEAL_II_DEPRECATED
         BoundaryTemperatureType *
         find_boundary_temperature_model () const;
 
+        template <typename BoundaryTemperatureType>
```

Comment:
Please put a documenting comment in front of this function (just copy and replace the one from postprocessor/interface.h)